Badge documentation (#1956)
telerik/blazor-docs: components/badge/position.md

---
title: Position
page_title: Badge Position
description: Explore the position and alignment settings of the Badge for Blazor. The example in the bottom of the page lets you experiment with the available parameters.
slug: badge-position-alignment
tags: telerik,blazor,badge,position,align,alignment
published: True
position: 35
---

# Position Settings

> This article outlines the available Badge parameters, which control its position and alignment.

Reviewer: Try to provide some context at the beginning of the article.
```suggestion
You can position and align the Blazor Badge component relative to its parent container by using the available parameters.
```

## Position

The `Position` parameter accepts a member of the `BadgePosition` enum and controls where the badge displays based on the parent container:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Enum member | Description |
|---------------|--------|
| `Edge` <br /> (default) | The center of the Badge will touch the edge of the parent container. |
| `Inside` | The Badge renders entirely inside the parent container. |
| `Outside`   | The Badge renders entirely outside the parent container. | 

See the [example](#example) below to customize the available parameters and see how they affect the Badge component.

## Alignment

Use the available parameters to control which side of the Badge touches the parent container.

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Enum Members |
| ----------- | ----------- | ----------- |
| `HorizontalAlign` | `BadgeHorizontalAlign` | `Start` <br /> `End` (default value) |
| `VerticalAlign` | `BadgeVerticalAlign` | `Top` (default value) <br /> `Bottom` |

> See the [example](#example) below to customize the available parameters and see how they affect the Badge component.

Reviewer: ```suggestion
Refer to the [example](#example) below to customize the available parameters and observe their impact on the Badge component.
```

## Example

The following example lets you experiment with the available settings that control the position and alignment of the Badge. It starts with the default component behavior.

````CSHTML
<div class="container">
    <div class="row">
        <div class="col-md-4">
            <label>
                Position
                <TelerikDropDownList Data="@Positions" @bind-Value="@Position"></TelerikDropDownList>
            </label>
        </div>
        <div class="col-md-4">
            <label>

>                 Horizontal Position

Reviewer 2: ```suggestion
                Horizontal Alignment
```

                <TelerikDropDownList Data="@BadgeHorizontalAlignSettings" @bind-Value="@BadgeHorizontalAlignSetting"></TelerikDropDownList>
            </label>
        </div>
        <div class="col-md-4">
            <label>

>                 Vertical Position

Reviewer 2: ```suggestion
                Vertical Alignment
```

                <TelerikDropDownList Data="@BadgeVerticalAlignSettings" @bind-Value="@BadgeVerticalAlignSetting"></TelerikDropDownList>
            </label>
        </div>
    </div>

>     <div class="row">

Reviewer 2: Add some vertical space between the two rows. Currently the Button touches the middle DropDownList.
